File: Eclipse/w7cpframe/src/clsParameters.java
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
import java.io.*;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

    class clsParameters
    {

        public int njt = 0;        //.. No. of joints ..
        public int nmb = 0;        //.. No. of members ..
        public int nmg = 0;        //.. No. of material groups ..
        public int nsg = 0;        //.. No. of member section groups ..
        public int nrj = 0;        //.. No. of supported reaction joints ..
        public int njl = 0;        //.. No. of loaded joints ..
        public int nml = 0;        //.. No. of loaded members ..
        public int ngl = 0;        //.. No. of gravity load cases .. Self weight
        public int nr = 0;         //.. No. of restraints @ the supports ..
        public int mag = 0;        //.. Magnification Factor for graphics

        
        public void initialise()
        {
            System.out.println("initialise ...");
            njt = 0;
            nmb = 0;

            nrj = 0;
            nmg =0;

            nsg =0;
            njl = 0;

            nml = 0;
            ngl = 0;

            nr = 0;

            mag = 0;
            System.out.println("... initialise");
        }

        public String sprint()
        {
            String s = "";
            s = s + String.format("%6d",njt) + String.format("%6d",nmb);
            s = s + String.format("%6d",nrj) + String.format("%6d",nmg);
            s = s + String.format("%6d",nsg) + String.format("%6d",njl);
            s = s + String.format("%6d",nml) + String.format("%6d",ngl);
            s = s + String.format("%6d",mag);

            return s;
        }

        public void cprint()
        {
            System.out.println(sprint());
        }

        public void fprint(BufferedWriter fp)
        {
            try {
				fp.write(sprint());
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
        }

        public void fgetData(BufferedReader fp, boolean isIgnore)
        {
            String s="";

            System.out.println("fgetData ...");

            try {
				s = fp.readLine();
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
            sgetData(s,isIgnore);

            System.out.println("... fgetData");
        }       
        
        public void sgetData(String s, boolean isIgnore)
        {
            String [] dataflds = new String[10]; //(0 To 9);
            int i;

            System.out.println("clsParameters:sgetData ...");

            //Trim Trailing Spaces from input String
            String regExpTrimPattern = "^\\s+|\\s+$"; //trim trailing spaces          
            Pattern regExpPatternObj = Pattern.compile(regExpTrimPattern);
            Matcher regExpMatches = regExpPatternObj.matcher(s);
            String regResult = regExpMatches.replaceAll("");
            
            //Get Data fields from input String
            String regExpNumberPattern = "-?\\d+(?:[,.]\\d+)?";
            regExpPatternObj = Pattern.compile(regExpNumberPattern);
            regExpMatches = regExpPatternObj.matcher(regResult);
            i=0;
            while (regExpMatches.find()){
            	dataflds[i] = regExpMatches.group().trim();
            	i = i+1;
            }

            //typically ignore as all counters are incremented as data read
            //isIgnore=False only used to test parser.
            if (isIgnore)
            {
                //Clear the control data, and count records as read data from file
                initialise(); // (0);
            }
            else
            { //Testing Reading Data

                njt = Integer.parseInt(dataflds[0]);
                nmb = Integer.parseInt(dataflds[1]);
                nrj = Integer.parseInt(dataflds[2]);
                nmg = Integer.parseInt(dataflds[3]);
                nsg = Integer.parseInt(dataflds[4]);
                njl = Integer.parseInt(dataflds[5]);
                nml = Integer.parseInt(dataflds[6]);
                ngl = Integer.parseInt(dataflds[7]);

            }


            if (dataflds[8] != "")
            {
                mag = Integer.parseInt(dataflds[8]);
            }
            else
            {
                mag = 1;
            }

            System.out.println("Dimension & Geometry");
            System.out.println("-------------------------------");
            System.out.format("Number of Joints       : %d", njt);
            System.out.format("Number of Members      : %d", nmb);
            System.out.format("Number of Supports     : %d", nrj);

            System.out.println("Materials & Sections");
            System.out.println("-------------------------------");
            System.out.format("Number of Materials    : %d", nmg);
            System.out.format("Number of Sections     : %d", nsg);

            System.out.println("Design Actions");
            System.out.println("-------------------------------");
            System.out.format("Number of Joint Loads  : %d", njl);
            System.out.format("Number of Member Loads : %d", nml);
            System.out.format("Number of Gravity Loads : %d", ngl);

            System.out.format("Screen Magnifier: %d", mag);

            System.out.println("... clsParameters:sgetData");
        }



    } //Class

File: VisualStudio/VBasic/wgpframe/clsParameters.vb
'
' Copyright (c)2016 S C Harrison
' Refer to License.txt for terms and conditions of use.
'
Public Class clsParameters

    Public njt As Integer        '.. No. of joints ..
    Public nmb As Integer        '.. No. of members ..
    Public nmg As Integer        '.. No. of material groups ..
    Public nsg As Integer        '.. No. of member section groups ..
    Public nrj As Integer        '.. No. of supported reaction joints ..
    Public njl As Integer        '.. No. of loaded joints ..
    Public nml As Integer        '.. No. of loaded members ..
    Public ngl As Integer        '.. No. of gravity load cases .. Self weight
    Public nr As Integer        '.. No. of restraints @ the supports ..

    Public mag As Integer     '.. Magnification Factor for graphics

Function getInitValue(baseIndx As Integer)
  If baseIndx = 0 Then
    getInitValue = -1
  ElseIf baseIndx = 1 Then
    getInitValue = 0
  End If
End Function

Sub initialise()
        Console.WriteLine("initialise ...")
        'njt = getInitValue(baseIndex)
        'nmb = getInitValue(baseIndex)
        'nmg = getInitValue(baseIndex)
        'nsg = getInitValue(baseIndex)
        'nrj = getInitValue(baseIndex)
        'njl = getInitValue(baseIndex)
        'nml = getInitValue(baseIndex)
        'ngl = getInitValue(baseIndex)
        'nr = getInitValue(baseIndex)

        'Total Count NOT indices
        'But used as array index during reading
        njt = 0
        nmb = 0
        nmg = 0
        nsg = 0
        nrj = 0
        njl = 0
        nml = 0
        ngl = 0
        nr = 0


        Console.WriteLine("... initialise")
End Sub

Function sprint() As String
  sprint = StrLPad(Format(njt, "0"), 6) & StrLPad(Format(nmb, "0"), 6) _
          & StrLPad(Format(nrj, "0"), 6) & StrLPad(Format(nmg, "0"), 6) _
          & StrLPad(Format(nsg, "0"), 6) & StrLPad(Format(njl, "0"), 6) _
          & StrLPad(Format(nml, "0"), 6) & StrLPad(Format(ngl, "0"), 6) _
          & StrLPad(Format(mag, "0"), 6)

End Function

Sub cprint()
        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & sprint
End Sub

Sub fprint(fp As Integer)
        Print(fp, sprint)
End Sub

    Sub fgetData(ByVal fp As Integer, ByVal isIgnore As Boolean)

        Dim s As String
        Dim n As Integer
        Dim dataflds(10) As String
        'Dim i As Integer

        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "clsParameters.fgetData ..."

        s = Trim(Readln(fp))
        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "<" + s + ">"

        Call parseDelimitedString(s, dataflds, n, " ")

        'typically ignore as all counters are incremented as data read
        'isIgnore=False only used to test parser.
        If isIgnore Then
            'Clear the control data, and count records as read data from file
            frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "Ignore Control Variable: Count as read data"
            initialise()
        Else
            frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "Reading Control Variables from File"

            'For i = 0 To n
            '    frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & dataflds(i)
            'Next

            njt = CInt(dataflds(0))
            nmb = CInt(dataflds(1))
            nrj = CInt(dataflds(2))
            nmg = CInt(dataflds(3))
            nsg = CInt(dataflds(4))
            njl = CInt(dataflds(5))
            nml = CInt(dataflds(6))
            ngl = CInt(dataflds(7))
        End If


        If dataflds(8) <> "" Then
            mag = CInt(dataflds(8))
        Else
            mag = 1
        End If

        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "Dimension & Geometry"
        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "-------------------------------"
        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "Number of Joints       : " + Format(njt, "0")
        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "Number of Members      : " + Format(nmb, "0")
        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "Number of Supports     : " + Format(nrj, "0")

        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "Materials & Sections"
        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "-------------------------------"
        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "Number of Materials    : " + Format(nmg, "0")
        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "Number of Sections     : " + Format(nsg, "0")

        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "Design Actions"
        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "-------------------------------"
        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "Number of Joint Loads  : " + Format(njl, "0")
        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "Number of Member Loads : " + Format(nml, "0")
        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "Number of Gravity Loads : " + Format(ngl, "0")

        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "Screen Magnifier: " + Format(mag, "0")

        frmTracer.txtTracer.Text = frmTracer.txtTracer.Text & vbCrLf & "... clsParameters.fgetData"
    End Sub

    '#If isUsingExcel = 1 Then

    '  Sub wbkPrint(dataSht As Worksheet)

    '    With dataSht
    '      .Range("Njoints").Value = njt
    '      .Range("Nmembers").Value = nmb
    '      .Range("Nmaterials").Value = nmg
    '      .Range("Nsections").Value = nsg

    '      .Range("Nsupports").Value = nrj


    '      .Range("Njloads").Value = njl
    '      .Range("Nmloads").Value = nml
    '      .Range("Ngloads").Value = ngl

    '      .Range("Mag").Value = mag
    '    End With

    '  End Sub

    '  Sub wbkRead(dataSht As Worksheet, isIgnore As Boolean)

    '    If isIgnore Then
    '      'Clear the control data, and count records as read data from file
    '      initialise
    '    Else
    '      With dataSht
    '        njt = .Range("Njoints").Value
    '        nmb = .Range("Nmembers").Value
    '        nmg = .Range("Nmaterials").Value
    '        nsg = .Range("Nsections").Value
    '        nrj = .Range("Nsupports").Value
    '        njl = .Range("Njloads").Value
    '        nml = .Range("Nmloads").Value
    '        ngl = .Range("Ngloads").Value
    '      End With
    '    End If

    '    mag = dataSht.Range("Mag").Value

    '  End Sub

    '#End If


End Class
